app/verify/get-files.js

```diff
@@ -0,0 +1,14 @@
+const retry = require('../retry')
+const storage = require('../storage')
+
+const getFiles = async (pendingFilenames) => {
+  // ensure we also have a control file for checksum before continuing
+  await retry(() => storage.getFile(pendingFilenames.checksumControlFilename))
```

Comment:
Tech debt perhaps: could we use `Joi` as a more universal and understood way of verifying the filenames are expected

Reply:
aaahhh we need to check if it's there too, mmhhh...

Reply:
Yes.  Made slightly more challenging as publishing to blob storage is asynchronous.  So even if ATOS transfer the files in a specified order, doesn't mean they'll arrive in that order and doesn't mean that they'll all be available at the same time.

So rather than throw an error immediately for a missing file, we build in a retry so it makes several attempts to get the file before ultimately throwing an error.

It of course will naturally retry on the next polling run, but building a retry in saves our logs being flooded with "file not found" errors unnecessarily.

Related, we have similar in the `ffc-pay-file-sender/ffc-pay-file-publisher` and `ffc-pay-statement-publisher` where we have one service that writes a file to storage and then sends a message to service bus instructing those services to do something with it.   The service bus message often arrives before the file is available in blob storage, so you'll see simliar retry patterns there.